lib/app/features/feed/create_story/views/pages/story_record_page.dart

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/components/progress_bar/centered_loading_indicator.dart';
+import 'package:ion/app/components/screen_offset/screen_bottom_offset.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/core/permissions/data/models/permissions_types.dart';
+import 'package:ion/app/features/core/permissions/providers/permissions_provider.dart';
+import 'package:ion/app/features/core/permissions/views/components/permission_aware_widget.dart';
+import 'package:ion/app/features/core/permissions/views/components/permission_dialogs/permission_sheets.dart';
+import 'package:ion/app/features/feed/create_story/hooks/use_camera_recording_controller.dart';
+import 'package:ion/app/features/feed/create_story/hooks/use_recording_progress.dart';
+import 'package:ion/app/features/feed/create_story/providers/story_camera_provider.dart';
+import 'package:ion/app/features/feed/create_story/views/components/story_camera/components.dart';
+import 'package:ion/app/features/gallery/providers/camera_provider.dart';
+import 'package:ion/app/router/app_routes.dart';
+
+class StoryRecordPage extends HookConsumerWidget {
+  const StoryRecordPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final storyCameraState = ref.watch(storyCameraControllerProvider);
+
+    useOnAppLifecycleStateChange((_, AppLifecycleState current) async {
```

Comment:
We have a `AppLifecycleProvider` now, I suggest you to migrate and use it inside `cameraControllerNotifierProvider`. The issue is that you might forget to pause/resume camera if you use the provider in multiple places.

Reply:
yeah, makes sense

Reply:
done